### ISSUE TITLE
nfsserver RA can't detect the failure of rpc.mountd on systemd. 

### ISSUE THREAD

**Issue by ytakeshita** [ClusterLabs/resource-agents#835]
Hi, All

On RHEL 7 with systemd, When rpc.mountd is dead, the NFS server cluster is service outage.
The reason for this is that both nfs-server.service and nfsserver RA doesn't monitor rpc.mountd. 
(In case of init script, the rpc.mountd is monitored by the script, so nfsserver RA can detect the failure.) 

I think that it is necessary to manage the rpc.mountd in nfsserver RA.
(However this depend on the unit files, so this may not occur in other distributions.)

Do you have any idea ?

To resolve this problem without fixing nfsserver RA, I have attempted to manage rpc.mountd
 as a resource in the following 2 patterns.
However both doesn't work well.

* pattern 1: insert`before` nfsserver
```
    Resource Group: grpNFS
         prmExNFS   (ocf::heartbeat:sfex):  Started mcha1
         prmFsNFS   (ocf::heartbeat:Filesystem):    Started mcha1
         prmIpNFS   (ocf::heartbeat:IPaddr2):       Started mcha1
         prmNFSmountd       (systemd:nfs-mountd):   Started mcha1
         prmNFS     (ocf::heartbeat:nfsserver):     Started mcha1
         prmInitNFSidmapd   (systemd:nfs-idmapd):   Started mcha1
```
When prmNFSmountd resource (nfs-mountd.service) started, nfs-server service is started together
 by dependency of nfs-mountd.service. 
`BindsTo=nfs-server.service`
So begining of nfsserver_start() return $OCF_SUCCESS without executing bind_tree () and so on.
For this reason the NFS server doesn't take over some information after failover.

* pattern 2: insert `after` nfsserver
```
    Resource Group: grpNFS
         prmExNFS   (ocf::heartbeat:sfex):  Started mcha1
         prmFsNFS   (ocf::heartbeat:Filesystem):    Started mcha1
         prmIpNFS   (ocf::heartbeat:IPaddr2):       Started mcha1
         prmNFS     (ocf::heartbeat:nfsserver):     Started mcha1
         prmNFSmountd       (systemd:nfs-mountd):   Started mcha1
         prmInitNFSidmapd   (systemd:nfs-idmapd):   Started mcha1
```
This is better than pattern 1.
In this case, grpNFS works well until ACT is stopped.
When prmNFSmountd resource (nfs-mount.service) stopped, the nfs-server service is stopped together
 by dependency of nfs-server.service. 
So, if nfsserver_monitor () is executed before nfsserver_stop (), prmNFS's monitor error (not running) is occurred. 

regerds,

**Comment by oalbrigt** [comment]
This patch should solve the issue: https://github.com/ClusterLabs/resource-agents/pull/836

**Comment by ytakeshita** [comment]
I have confiremed that the problem has been modified with your patch.
Thank you very much. Close this issue.
But I find 2 problems, and fix them PR #842
